controllers/machine_controller.go

```diff
@@ -471,16 +457,16 @@ func (r *MachineReconciler) isDeleteNodeAllowed(ctx context.Context, cluster *cl
 		}
 	}
 
-	// Get all of the machines that belong to this cluster.
-	machines, err := getActiveMachinesInCluster(ctx, r.Client, machine.Namespace, machine.Labels[clusterv1.ClusterLabelName])
+	// Get all of the active machines that belong to this cluster.
+	machines, err := collections.GetFilteredMachinesForCluster(ctx, r.Client, cluster, collections.ActiveMachines)
 	if err != nil {
 		return err
 	}
 
 	// Whether or not it is okay to delete the NodeRef depends on the
 	// number of remaining control plane members and whether or not this
 	// machine is one of them.
-	switch numControlPlaneMachines := len(util.GetControlPlaneMachines(machines)); {
+	switch numControlPlaneMachines := len(machines.Filter(collections.ControlPlaneMachines(cluster.Name))); {
```

Comment:
Would it make sense to simplify this to an `if / else` statement? Would be a bit easier to read for me at least with only 2 outcomes.